prediction_market_agent_tooling/markets/seer/seer_subgraph_handler.py

```diff
@@ -123,43 +127,46 @@ def _build_where_statements(
         if not include_conditional_markets:
             and_stms["parentMarket"] = ADDRESS_ZERO.lower()
 
-        yes_stms, no_stms = {}, {}
-        exclude_scalar_yes, exclude_scalar_no = {}, {}
+        outcome_filters: list[dict[str, t.Any]] = []
 
-        # Return scalar markets.
-        if include_only_scalar_markets:
-            # We are interested in scalar markets only - this excludes categorical markets
-            yes_stms = SeerSubgraphHandler._create_case_variations_condition(
+        if market_type == MarketType.SCALAR:
+            # Template ID "1" + UP/DOWN outcomes for scalar markets
+            and_stms["templateId"] = 1
+            up_filter = SeerSubgraphHandler._create_case_variations_condition(
                 UP_OUTCOME_LOWERCASE_IDENTIFIER, "outcomes_contains", "or"
             )
-            no_stms = SeerSubgraphHandler._create_case_variations_condition(
+            down_filter = SeerSubgraphHandler._create_case_variations_condition(
                 DOWN_OUTCOME_LOWERCASE_IDENTIFIER, "outcomes_contains", "or"
             )
-        elif include_conditional_markets and not include_categorical_markets:
-            # We are interested in binary markets only
-            yes_stms = SeerSubgraphHandler._create_case_variations_condition(
+            outcome_filters.extend([up_filter, down_filter])
+
+        elif market_type == MarketType.BINARY:
+            # Template ID "2" + YES/NO outcomes for binary markets
+            and_stms["templateId"] = 2
+            yes_filter = SeerSubgraphHandler._create_case_variations_condition(
                 YES_OUTCOME_LOWERCASE_IDENTIFIER, "outcomes_contains", "or"
             )
-            no_stms = SeerSubgraphHandler._create_case_variations_condition(
+            no_filter = SeerSubgraphHandler._create_case_variations_condition(
                 NO_OUTCOME_LOWERCASE_IDENTIFIER, "outcomes_contains", "or"
             )
+            outcome_filters.extend([yes_filter, no_filter])
 
-        if (
-            not include_only_scalar_markets
-            or include_categorical_markets
-            or include_conditional_markets
-        ):
-            # We should not provide any scalar markets because they are exclusive for categorical markets
-            exclude_scalar_yes = SeerSubgraphHandler._create_case_variations_condition(
-                UP_OUTCOME_LOWERCASE_IDENTIFIER, "outcomes_not_contains", "and"
-            )
-            exclude_scalar_no = SeerSubgraphHandler._create_case_variations_condition(
-                DOWN_OUTCOME_LOWERCASE_IDENTIFIER, "outcomes_not_contains", "and"
+        elif market_type == MarketType.CATEGORICAL:
+            # Template ID 2 OR Template ID 3 (includes both binary and categorical)
```

Comment:
```suggestion
            # Template ID 2 (categorical) OR Template ID 3 (multi-categorical, we treat them as categorical for now for simplicity)
```

See https://reality.eth.limo/app/docs/html/contracts.html#templates